lib/matplotlib/ticker.py

```diff
@@ -1799,6 +1806,14 @@ def __call__(self):
         return self.tick_values(vmin, vmax)
 
     def tick_values(self, vmin, vmax):
+        if self.numticks == 'auto':
+            if self.axis is not None:
+                numticks = max(min(self.axis.get_tick_space(), 9), 2)
```

Comment:
You are right, it is pretty random.  It is the default max we chose for the MaxNLocator auto algorithm, so I used it here as well.  The rationale in both cases is that plots just don't look good with a huge number of major ticks; so 9 is random, but it "seems reasonable" as a limit.  Nothing prevents a user from specifying a larger number instead of using the auto algorithm.